portal/worker/src/walrus-sites-sw.ts

```diff
@@ -48,16 +49,39 @@ self.addEventListener("fetch", async (event) => {
     console.log("Portal domain and request domain: ", portalDomain, requestDomain);
     console.log("Parsed URL: ", parsedUrl);
 
-    if (requestDomain == portalDomain && parsedUrl && parsedUrl.subdomain) {
-        let response: Promise<Response>;
-        if (!("caches" in self)) {
-            // When not being in a secure context, the Cache API is not available.
-            console.warn("Cache API not available");
-            response = resolveAndFetchPage(parsedUrl);
-        } else {
-            response = resolveWithCache(parsedUrl, urlString);
-        }
-        event.respondWith(response);
+    if (requestDomain === portalDomain && parsedUrl && parsedUrl.subdomain) {
+        // Fetches the page resources.
+        const handleFetchRequest = async (): Promise<Response> => {
```

Comment:
Suggestion: separate this out a bit for readability?

Reply:
Done!